engine/servermaster/executor_manager.go

```diff
@@ -39,16 +43,18 @@ func init() {
 // ExecutorManager defines an interface to manager all executors
 type ExecutorManager interface {
 	HandleHeartbeat(req *pb.HeartbeatRequest) (*pb.HeartbeatResponse, error)
-	AllocateNewExec(req *pb.RegisterExecutorRequest) (*model.NodeInfo, error)
-	RegisterExec(info *model.NodeInfo)
+	AllocateNewExec(ctx context.Context, req *pb.RegisterExecutorRequest) (*ormModel.Executor, error)
 	// ExecutorCount returns executor count with given status
 	ExecutorCount(status model.ExecutorStatus) int
 	HasExecutor(executorID string) bool
-	ListExecutors() []*model.NodeInfo
+	ListExecutors() []*ormModel.Executor
 	GetAddr(executorID model.ExecutorID) (string, bool)
 	Start(ctx context.Context)
 	Stop()
 
+	// ResetExecutors reset all executors with the latest meta from database.
+	ResetExecutors(ctx context.Context) error
+
```

Comment:
Maybe this should be part of the initialization logic instead of part of the interface?

Reply:
Good idea!

Reply:
Addressed in https://github.com/pingcap/tiflow/pull/6918/commits/e5d1fc51f98a045215e3edfd1a67a3efe0e93733.